cdap-data-fabric/src/main/java/co/cask/cdap/data2/transaction/Transactions.java

```diff
@@ -427,5 +430,47 @@ private static TransactionControl getTransactionControl(Class<?> cls, String met
     return null;
   }
 
+  /**
+   * Executes the given callable with a transaction.
+   * @param transactional the {@link Transactional} used to submit the task
+   * @param callable the task
+   * @param <V> type of result
+   * @return value returned by the given {@link TxCallable}.
+   * @throws ServiceUnavailableException when the transaction service is not running
+   * @throws RuntimeException for any other errors that occurs
+   */
+  public static <V> V txExecute(Transactional transactional, TxCallable<V> callable) {
+    try {
+      return Transactions.execute(transactional, callable);
+    } catch (TransactionFailureException e) {
+      throw Transactions.propagate(e);
+    } catch (RuntimeException e) {
+      Throwable t = e.getCause();
+      if (t != null && t instanceof TException) {
+        throw new ServiceUnavailableException(Constants.Service.TRANSACTION, e);
+      }
+      throw e;
+    }
+  }
 
+  /**
+   * Executes the given runnable with a transaction.
+   * @param transactional the {@link Transactional} used to submit the task
+   * @param runnable task
+   * @throws ServiceUnavailableException when the transaction service is not running
+   * @throws RuntimeException for any other errors that occurs
+   */
+  public static void txExecute(Transactional transactional, TxRunnable runnable) {
+    try {
+      transactional.execute(runnable);
```

Comment:
to remove code duplication, you can create a new TxCallable from the TxRunnable and call the other method